git-branchless/tests/command/test_reword.rs

```diff
@@ -49,7 +49,7 @@ fn test_reword_current_commit_not_head() -> eyre::Result<()> {
     let (stdout, _stderr) = git.run(&["smartlog"])?;
     insta::assert_snapshot!(stdout, @r###"
     :
-    @ 62fc20d (test1) create test1.txt
+    @ 62fc20d (> test1) create test1.txt
```

Comment:
I think this is happening because `git reword` (and other rebasing-related functions?) happens to implicitly check out the current commit after it's done. Maybe we can update `check_out_commit` to only check out the branch if the target isn't the current commit? Or even skip the rest of the function body altogether?

Reply:
Oh, I guess that this is happening because the commit *is* different after the reword, i.e. it has a different commit hash. I guess we could work around this by looking at the old and new commits and seeing if the new version of the commit is a rewrite of the old one, and, if so, inhibiting the auto-check-out behavior.

That being said, I think it's a lot of complexity, and I don't envision users needing to do that. I expect that either they'll want to always have the one branch associated with the current commit checked out (and can use e.g. `git record --detach` to create a commit without affecting the current branch), or they would like the previous behavior where we just don't adjust the current branch.

I think the rest of this PR is good to ship. Could you add a configuration option to enable this behavior (we can enable it by default) and a test to confirm that the behavior doesn't happen when the behavior is disabled?

- Update the Wiki here: https://github.com/arxanas/git-branchless/wiki/Configuration
- Add the configuration option to this file: https://github.com/arxanas/git-branchless/blob/605d5dc641687cd87dfa41716ad8a005fc5aa9b6/git-branchless-lib/src/core/config.rs

I guess for the configuration variable name, we could call it something like `branchless.navigation.autoSwitchBranches`, or maybe you have a better idea? (Adopting the new Git terminology of "switch" for branches, since "check out" is overloaded.)

(There's also the existing configuration variable `branchless.next.interactive`, but I think that is a misnomer, and we should eventually move it to this new `branchless.navigation` section, because it also affects the `git prev` command.)

Reply:
Added the `branchless.navigation.autoSwitchBranches` configuration option, updated the documentation, and added two tests, one with the functionality turned on, and the other with it turned off.